app/views/course/assessment/submission/submissions/_statistics.html.slim

```diff
@@ -1,3 +1,4 @@
+- display_grading_results = @submission.published? || can?(:grade, @submission)
```

Comment:
Would it be better to place this as part of the cancancan ability? So it would be something along the lines of `allow_students_to_see_published_submission_grade`.